### PR TITLE
(Draft) Finer control for SponsorBlock

### PR DESCRIPTION
**Pull request by photochlorination**
I'm not sure if I'll (be able to) finish this, but I thought I'd make a draft to see if this is considered feasible.

- [x] Preference page selection
- [x] Update sponsor fetching code
- [x] Show-only segments
- [x] Manual skip
- [ ] Manual skip popup
- [ ] Migration of preferences
- [ ] Vue componentization (broken attempt at https://github.com/photochlorination/Piped/tree/sponsorblock-vue-component)

### CODE REVIEW

File: src/components/PreferencesPage.vue [hunk 1]
```diff
@@ -185,63 +185,84 @@
     </label>
     <label class="pref" for="chkSkipSponsors">
         <strong v-t="'actions.skip_sponsors'" />
-        <input id="chkSkipSponsors" v-model="skipSponsor" class="checkbox" type="checkbox" @change="onChange($event)" />
+        <select id="chkSkipSponsors" v-model="skipSponsor" @change="onChange($event)">
+            <option value="off">Disable</option>
+            <option value="skip">Auto Skip</option>
+            <option value="manual">Manual Skip</option>
+            <option value="show">Show</option>
+        </select>
```
Comment (FireMasterK) [inline src/components/PreferencesPage.vue:193]: Rather than duplicating this code, please turn it into a component!

**Review by FireMasterK — CHANGES_REQUESTED** [review]

File: src/components/PreferencesPage.vue (hunk 1 above)
Comment (photochlorination) [reply src/components/PreferencesPage.vue:193]: I'm not familiar with Vue, do you have any tips on how to do this?  I tried moving the select element into a component but it broke the preference value being saved/loaded.

Comment (Bnyro) [reply src/components/PreferencesPage.vue:193]: You might want to have a look at the `WatchOnYouTubeButton` which is used as a component, in the end components just work like normal Vue pages/files!

Comment (photochlorination) [reply src/components/PreferencesPage.vue]: I think I have it kind of figured out, but I still can't figure out how to do the onChange portion.

Comment (FireMasterK) [reply src/components/PreferencesPage.vue:193]: Ideally, I would move that portion of the code inside the component too, I could help with that by pushing into this branch too

Comment (photochlorination) [reply src/components/PreferencesPage.vue:193]: I'd be thankful for any help getting it to work, this is pretty much my first time working with Vue so I don't really know what I'm doing.

**Comment by photochlorination** [comment]
Apologies for abandoning this draft, but I've given up on the UI portion. I won't be completing this, but if anyone else is interested to they are free to use the code.

**Comment by trymeouteh** [comment]
Would like to have the option for manual skip and auto skip

**Pull request closed without merging**